### ISSUE TITLE
Divide and Conquer the Embedding Space for Metric Learning

### ISSUE THREAD

**Issue by juanmc2005** [juanmc2005/phd-bib#2]
## [Divide and Conquer the Embedding Space for Metric Learning](https://arxiv.org/abs/1906.05990)
*Learning the embedding space, where semantically similar objects are located close together and dissimilar objects far apart, is a cornerstone of many computer vision applications. Existing approaches usually learn a single metric in the embedding space for all available data points, which may have a very complex non-uniform distribution with different notions of similarity between objects, e.g. appearance, shape, color or semantic meaning. Approaches for learning a single distance metric often struggle to encode all different types of relationships and do not generalize well. In this work, we propose a novel easy-to-implement divide and conquer approach for deep metric learning, which significantly improves the state-of-the-art performance of metric learning. Our approach utilizes the embedding space more efficiently by jointly splitting the embedding space and data into K smaller sub-problems. It divides both, the data and the embedding space into K subsets and learns K separate distance metrics in the non-overlapping subspaces of the embedding space, defined by groups of neurons in the embedding layer of the neural network. The proposed approach increases the convergence speed and improves generalization since the complexity of each sub-problem is reduced compared to the original one. We show that our approach outperforms the state-of-the-art by a large margin in retrieval, clustering and re-identification tasks on CUB200-2011, CARS196, Stanford Online Products, In-shop Clothes and PKU VehicleID datasets.*

### Cite this paper
```bibtex
@article{DBLP:journals/corr/abs-1906-05990,
  author    = {Artsiom Sanakoyeu and
               Vadim Tschernezki and
               Uta B{\"{u}}chler and
               Bj{\"{o}}rn Ommer},
  title     = {Divide and Conquer the Embedding Space for Metric Learning},
  journal   = {CoRR},
  volume    = {abs/1906.05990},
  year      = {2019},
  url       = {http://arxiv.org/abs/1906.05990},
  archivePrefix = {arXiv},
  eprint    = {1906.05990},
  timestamp = {Mon, 24 Jun 2019 17:28:45 +0200},
  biburl    = {https://dblp.org/rec/bib/journals/corr/abs-1906-05990},
  bibsource = {dblp computer science bibliography, https://dblp.org}
}
```

**Comment by juanmc2005** [comment]
This paper begins by explaining that one of the biggest problems of metric learning is that it has a one-fits-all approach in which a single projection function is trained to cluster embeddings in the target space. However, examples can vary in different ways, making this approach very difficult to optimize.
Even though it has been proven that neural networks are universal approximators, the optimization of a single projection function is more difficult than the optimization of multiple simpler ones, because of poor local minima and overfitting.

The authors propose an approach for image metric learning where K different projection sub-functions are learned in a subspace of the target space.
The architecture is composed of an encoder that maps the images into the target space (optimized using triplet loss), a K-means clustering stage on the embeddings, and one learner for each of the identified subspaces (also trained with triplet loss). Finally, all K sub-embeddings are concatenated to form the final representation of the image. The trunk CNN architecture is the same for every learner and the initial encoder.

Since the initial embeddings change over time, K-means clustering is updated every T epochs.
Sub-embedding learners are trained in an alternating manner, uniformly sampling clusters and drawing a mini-batch from it.

![Proposed model](https://raw.githubusercontent.com/juanmc2005/phd-bib/master/sanakoyeu_etal2019.png)

When training is finished, a fine tuning of the initial encoder is needed to achieve consistency between the embeddings of individual learners.

They achieve (and in some cases surpass) state of the art performance on many of their evaluation tasks (Stanford Online Products, CARS196, CUB200-2011, In-shop Clothes Retrieval and PKU VehicleID).
